app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -199,6 +252,169 @@ fun HomeScreen(
   }
 }
 
+@Composable
+fun SearchBarAndFilter(
+    searchQuery: String,
+    onSearchQueryChange: (String) -> Unit,
+    onSearchButtonClick: () -> Unit,
+    showFilterPopUp: Boolean,
+    setShowFilterPopUp: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  Row(verticalAlignment = Alignment.CenterVertically, modifier = modifier) {
+    // Search bar
+    TextField(
+        value = searchQuery,
+        // function onSearchQueryChange() needs to be created in the VM
+        onValueChange = { onSearchQueryChange(it) },
+        modifier = Modifier.weight(1f),
+        maxLines = 1,
+        shape = RoundedCornerShape(32.dp),
+        colors =
+            TextFieldDefaults.colors(
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                disabledIndicatorColor = Color.Transparent),
+        placeholder = { Text("Search event...") },
+        trailingIcon = {
+          IconButton(
+              onClick = onSearchButtonClick, modifier = Modifier.padding(horizontal = 4.dp)) {
+                Icon(Icons.Default.Search, contentDescription = null)
+              }
+        })
+
+    // Filter button
+    Button(
+        // function setShowFilterPopUp() needs to be created in the VM
+        onClick = { setShowFilterPopUp(!showFilterPopUp) },
+        modifier = Modifier.padding(start = 8.dp)) {
+          Text("Filter")
+        }
+  }
+}
+
+@Composable
+fun FilterPopUp(
+    onRadiusChange: (Double) -> Unit,
+    onFreeSelectionChange: (Boolean) -> Unit,
+    onCategorySelectionChange: (List<EventCategory>) -> Unit,
+    modifier: Modifier = Modifier,
+    uiState: EventsOverviewUiState = EventsOverviewUiState()
```

Comment:
You should not create a state like this, but rather derive the state from the view model using `collectAsState` (or `collectAsStateWithLifecycle` in the example below)
![Screenshot 2024-04-29 154033](https://github.com/swent-sp-2024-event-radar/event-radar-app/assets/44499915/0c69fbe9-8621-4efb-9690-940784fa9dec)

Reply:
Line changed to:
    uiState: EventsOverviewUiState,

As assigning state caused bugs.